packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java

```diff
@@ -12,7 +12,30 @@
 import java.io.IOException;
 
 class ImageResizer {
-  File resizedImage(String path, Double maxWidth, Double maxHeight) throws IOException {
+  private final ExifDataCopier exifDataCopier;
+
+  ImageResizer(ExifDataCopier exifDataCopier) {
+    this.exifDataCopier = exifDataCopier;
+  }
+
+  String resizeImageIfNeeded(String imagePath, Double maxWidth, Double maxHeight) {
+    boolean shouldScale = maxWidth != null || maxHeight != null;
+
+    if (!shouldScale) {
+      return imagePath;
+    } else {
+      try {
+        File scaledImage = resizedImage(imagePath, maxWidth, maxHeight);
+        exifDataCopier.copyExif(imagePath, scaledImage.getPath());
+
+        return scaledImage.getPath();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
Why not just bubble up the more specific IOException?

Reply:
Since `IOException` is a checked exception, throwing it would require to change this method to look like this:

```java
String resizeImageIfNeeded(String imagePath, Double maxWidth, Double maxHeight) throws IOException {
}
```

Which would require the calling code (lines 221-> in `ImagePickerDelegate`) to handle the exception like this:

```java
private void handleResult(String path) {
  ...
  try {
    String finalImagePath = imageResizer.resizeImageIfNeeded(path, maxWidth, maxHeight);
    finishWithSuccess(finalImagePath);
  } catch (IOException e) {
    e.printStackTrace();
  }
}
```

I think it's a matter of preference, but personally favor the `ImageResizer` to swallow it's own exceptions, but I'm fine with either way.